.circleci/config.yml

```diff
@@ -1,21 +1,148 @@
+          command: |
+            set -x
+            go_ver=$(grep -r '^go ' go.mod | cut -d ' ' -f2)
+            test -n "$go_ver"
+            working_dir=/home/circleci/project
+            touch env_vars.sh
+            echo "export WORKING_DIR=$working_dir" >> env_vars.sh
+            echo "export GO_VERSION=$go_ver" >> env_vars.sh
+      - persist_to_workspace:
+          root: .
+          paths:
+            - env_vars.sh
+  build_binaries:
+    machine:
+      enabled: true
+      docker_layer_caching: true
+    steps:
+      - attach_workspace:
+          at: /tmp/workspace
+      - source-vars
+      - checkout
+      - run:
+          name: Build source tarballs
+          command: |
+            set -x
+            mkdir -p ${WORKING_DIR}/tarball
+            "${WORKING_DIR}/releng/source-tarball/build.bash" \
+              -p "${WORKING_DIR}/.git" \
+              -s "${CIRCLE_SHA1}" \
+              -b "${CIRCLE_BRANCH}" \
+              -v "${CIRCLE_BRANCH}-${CIRCLE_SHA1}" \
+              -o "${WORKING_DIR}/tarball"
+      - store_artifacts:
+          path: tarball/
+      - run:
+          name: Build binaries
+          command: |
+            OUTDIR=${WORKING_DIR}/bins
+            mkdir -p ${OUTDIR}
+            GOOS=linux GOARCH=amd64 "${WORKING_DIR}/releng/raw-binaries/build.bash" \
+              -i "${WORKING_DIR}/tarball/influxdb-src-${CIRCLE_SHA1}.tar.gz" \
+              -o "$OUTDIR"
+      - store_artifacts:
+          path: bins/
+      - run:
+          name: Build packages
+          command: |
+            OUTDIR=${WORKING_DIR}/oss-packages
+            BINDIR=${WORKING_DIR}/bins
+            mkdir -p ${OUTDIR}
+            mkdir -p ${BINDIR}
+            "${WORKING_DIR}/releng/packages/build.bash" \
+              -s "${WORKING_DIR}/tarball/influxdb-src-${CIRCLE_SHA1}.tar.gz" \
+              -b "${BINDIR}/influxdb_bin_linux_amd64-${CIRCLE_SHA1}.tar.gz" \
+              -O linux -A amd64 \
+              -o "$OUTDIR"
+      - store_artifacts:
+          path: oss-packages/
+      - when:
+          condition:
+            and:
+              - equal: [ master-1.x, << pipeline.git.branch >> ]
+              - << pipeline.git.tag >>
+          steps:
+            - aws-s3/copy:
+                arguments: |
+                  --aws-region us-east-1
+                from: 'tarball'
+                to: 's3://dl.influxdata.com/influxdb/releases'
+            - aws-s3/copy:
+                arguments: |
+                  --aws-region us-east-1
+                from: 'bins'
+                to: 's3://dl.influxdata.com/influxdb/releases'
+            - aws-s3/copy:
+                arguments: |
+                  --aws-region us-east-1
+                from: 'oss-packages'
+                to: 's3://dl.influxdata.com/influxdb/releases'
+  changelog:
+    docker:
+      - image: jsternberg/changelog:latest
+    steps:
+      - checkout
+      - run:
+          name: changelog-magic
```

Comment:
Do we still want to do this?  Check with @lesam and @danxmoran; they may have wanted to eliminate the automated changelog stuff.

Reply:
Good question, I'd love to hear the answer.

Reply:
My vote would be to not do it on every merge, and instead have a script that can look at the commits between 2 tags and generate a full log from that. We'd then `rm` the `CHANGELOG.md` file from the git repo, which would save us from a lot of pointless merge conflicts when we cherry-pick commits between branches.

Reply:
I understand the automated changelog was not behaving 'properly' - @dgnorton might have details, it was before my time that it was disabled.

I would love to have changelogs that 'just worked' without manual effort.

Reply:
So a couple of things:

* the changelog job here will only run daily for branches matching the regex -- not on every push, and not for every branch
* the changelog magic that comes out of the jsternberg docker container is happening today in Jenkins for those 1.x releases that still run there